lib/good_result.rb

```diff
@@ -3,6 +3,45 @@
 require_relative "good_result/version"
 
 module GoodResult
-  class Error < StandardError; end
-  # Your code goes here...
+  class ResultError < StandardError; end
+  
+  class Success
+    attr_reader :value
+
+    def initialize(value)
+      @value = value
+    end
+
+    def success?
+      true
+    end
+
+    def failure?
+      false
+    end
+
+    def error
+      raise ResultError, 'Expected error but Result is Success'
+    end
+  end
+
+  class Failure
+    attr_reader :error
+
+    def initialize(error)
+      @error = error
+    end
+
+    def success?
+      false
+    end
+
+    def failure?
+      true
+    end
+
+    def value
+      raise ResultError, 'Expected value but Result is Failure'
+    end
```

Comment:
Basically the same as above. Only thing to add here is, I'm not entirely sure I understand the message here. Is it saying that `@error` is `nil`? Or just that this Result object is a Failure object?